test/objects.js

```diff
@@ -681,6 +681,67 @@
     assert.ok(_.isBoolean(false), 'and so is false');
   });
 
+  QUnit.test('isMap', function(assert) {
+    var keyString = 'a string', keyObj = {}, keyFunc = function() {};
+    var obj = new Map([[keyString, 'value one'], [keyObj, 'value two'],
+      [keyFunc, 'value three']]);
+
+    assert.ok(!_.isMap('string'), 'a string is not a map');
+    assert.ok(!_.isMap(2), 'a number is not a map');
+    assert.ok(!_.isMap({}), 'an object is not a map');
+    assert.ok(!_.isMap(false), 'a boolean is not a map');
+    assert.ok(!_.isMap(void 0), 'undefined is not a map');
+    assert.ok(!_.isMap([1, 2, 3]), 'an array is not a map');
+    assert.ok(!_.isMap(new Set([['key', 'value']])), 'a set is not a map');
+    assert.ok(!_.isMap(new WeakMap([[{x: 1}, 'value']])), 'a weakmap is not a map');
+    assert.ok(_.isMap(obj), 'but a map is');
+  });
+
+  QUnit.test('isWeakMap', function(assert) {
+    var keyObj = {};
+    var obj = new WeakMap([[keyObj, 'value']]);
+
+    assert.ok(!_.isWeakMap('string'), 'a string is not a weakmap');
+    assert.ok(!_.isWeakMap(2), 'a number is not a weakmap');
+    assert.ok(!_.isWeakMap({}), 'an object is not a weakmap');
+    assert.ok(!_.isWeakMap(false), 'a boolean is not a weakmap');
+    assert.ok(!_.isWeakMap(void 0), 'undefined is not a weakmap');
+    assert.ok(!_.isWeakMap([1, 2, 3]), 'an array is not a weakmap');
+    assert.ok(!_.isWeakMap(new Set([['key', 'value']])), 'a set is not a weakmap');
```

Comment:
The [same goes](https://github.com/jashkenas/underscore/pull/2464/files#r55551957) for the `Set` creation.